tests/test_batch_index_eu.py

```diff
@@ -0,0 +1,78 @@
+#    Copyright (C) 2020 Presidenza del Consiglio dei Ministri.
+#    Please refer to the AUTHORS file for more information.
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+#    GNU Affero General Public License for more details.
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+from datetime import timedelta
+
+import pytest
+from pytest_sanic.utils import TestClient
+
+from immuni_common.helpers.tests import mock_config
+from immuni_common.models.mongoengine.batch_file_eu import BatchFileEu
+from immuni_exposure_reporting.core import config
+from tests.fixtures.batch_file_eu import create_random_batches_eu
+
+
+async def test_index_no_batches_eu(client: TestClient) -> None:
+    response = await client.get("/v1/keys/eu/DK/index")
+    assert response.status == 404
+
+
+@mock_config(config, "MANIFEST_CACHE_TIME_IN_MINUTES", timedelta(minutes=30).total_seconds())
+@pytest.mark.parametrize("num_batches, oldest, newest", ((1, 0, 0), (10, 0, 9), (20, 6, 19)))
+async def test_index_eu(client: TestClient, num_batches: int, oldest: int, newest: int) -> None:
+    create_random_batches_eu(num_batches)
```

Comment:
There are only "DK" batches. IMHO, a proper test should create also other-origin batches too.